src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -1428,12 +1501,12 @@ fixCursorForBbfSqlvariantTableData( Archive *fout,
 			continue;
 
 		/*
-			* To find the basetype and bytelength of string data types we
-			* invoke sys.sql_variant_property and sys.datalength function on
-			* the sqlvariant column. These extra columns are added at the end of
-			* the select cursor query so that they do not interfere with
-			* expected dump behaviour.
-		*/
+		 * To find the basetype and bytelength of string data types we
+		 * invoke sys.sql_variant_property and sys.datalength function on
+		 * the sqlvariant column. These extra columns are added at the end of
+		 * the select cursor query so that they do not interfere with
+		 * expected dump behaviour.
+		 */
 		if (pg_strcasecmp(tbinfo->atttypnames[i],
 			quote_all_identifiers ? "\"sys\".\"sql_variant\"" : "sys.sql_variant") == 0)
 		{
```

Comment:
Let's have pre length check of the `atttypnames[i]` so that we compare the strings only when we have desired length:
```
size_t attlen = strlen(tbinfo->atttypnames[i]);
if (quote_all_identifiers ?
        (attlen == 19) && (pg_strcasecmp(tbinfo->atttypnames[i], "\"sys\".\"sql_variant\"") == 0)  :
        (attlen == 15) && (pg_strcasecmp(tbinfo->atttypnames[i], "sys.sql_variant") == 0))
else if (quote_all_identifiers ?
        (attlen == 21) && (pg_strcasecmp(tbinfo->atttypnames[i], "\"sys\".\"sql_variant\"[]") == 0)  :
        (attlen == 17) && (pg_strcasecmp(tbinfo->atttypnames[i], "sys.sql_variant[]") == 0))
```